ray-operator/config/samples/ray-service.tpu-single-host.yaml

```diff
@@ -0,0 +1,94 @@
+  name: s-d-tpu
+spec:
+  serveConfigV2: |
+    applications:
+      - name: stable_diffusion
+        import_path: stable_diffusion.stable_diffusion_tpu:deployment
+        runtime_env:
+          working_dir: "https://github.com/ryanaoleary/serve_config_examples/archive/refs/heads/tpu-example.zip"
+          pip:
+            - pydantic<2
+            - google-api-python-client
+            - pillow
+            - diffusers==0.7.2
+            - transformers==4.24.0
+            - flax
+            - ml_dtypes==0.2.0
+            - jax[tpu]==0.4.11
+            - -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+            - fastapi
+            - pillow
+            - ray[serve,default]==2.9.0
+  rayClusterConfig:
+    rayVersion: '2.9.0' # Should match the Ray version in the image of the containers
+    ######################headGroupSpecs#################################
+    # Ray head pod template.
+    headGroupSpec:
+      # The `rayStartParams` are used to configure the `ray start` command.
+      # See https://github.com/ray-project/kuberay/blob/master/docs/guidance/rayStartParams.md for the default settings of `rayStartParams` in KubeRay.
+      # See https://docs.ray.io/en/latest/cluster/cli.html#ray-start for all available options in `rayStartParams`.
+      rayStartParams:
+        dashboard-host: '0.0.0.0'
+      # Pod template
+      template:
+        spec:
+          containers:
+          - name: ray-head
+            image: rayproject/ray:2.9.0-py310
+            ports:
+            - containerPort: 6379
+              name: gcs
+            - containerPort: 8265
+              name: dashboard
+            - containerPort: 10001
+              name: client
+            - containerPort: 8000
+              name: serve
+            volumeMounts:
+              - mountPath: /tmp/ray
+                name: ray-logs
+            resources:
+              limits:
+                cpu: "2"
+                memory: "8G"
+              requests:
+                cpu: "2"
+                memory: "8G"
+          volumes:
+            - name: ray-logs
+              emptyDir: {}
+    workerGroupSpecs:
+    # The pod replicas in this group typed worker
+    - replicas: 1
+      minReplicas: 1
+      maxReplicas: 10
+      numOfHosts: 1
+      groupName: tpu-group
+      rayStartParams: {}
+      # Pod template
+      template:
+        spec:
+          containers:
+          - name: ray-worker
+            image: rayproject/ray:2.9.0-py310
+            resources:
+              limits:
+                cpu: "1"
+                ephemeral-storage: 20Gi
+                google.com/tpu: "4"
+                memory: 80G
+              requests:
+                cpu: "1"
+                ephemeral-storage: 20Gi
+                google.com/tpu: "4"
+                memory: 80G
+          nodeSelector:
+            cloud.google.com/gke-tpu-accelerator: tpu-v4-podslice
+            cloud.google.com/gke-tpu-topology: 2x2x1
```

Comment:
How can I determine whether a TPU PodSlice is single-host or multi-host based on `gke-tpu-topology`?

Reply:
This page has information about TPU topologies and whether they're single or multi host: https://cloud.google.com/kubernetes-engine/docs/concepts/tpus#autopilot-tpu-topology. In general, you can determine if a PodSlice is multi-host by taking the product of the topology (i.e. the total number of chips) and dividing it by the `google.com/tpu` value in the k8s_resource_limits (i.e. the number of TPU chips per Pod or VM host). In this example, the topology has 4 chips total and `google.com/tpu`=4, so it is single host.

Reply:
Added a link for this in [3b44939](https://github.com/ray-project/kuberay/pull/2198/commits/3b449390808f87876c499aacb31dac657b903191).